platform/mbed_power_mgmt.h

```diff
@@ -96,23 +96,32 @@ void sleep_tracker_unlock(const char *const filename, int line);
 /** Lock the deep sleep mode
  *
  * This locks the automatic deep mode selection.
+ * Verified by ::test_lock_unlock
+ *
  * sleep_manager_sleep_auto() will ignore deepsleep mode if
  * this function is invoked at least once (the internal counter is non-zero)
  *
  * Use this locking mechanism for interrupt driven API that are
  * running in the background and deepsleep could affect their functionality
  *
  * The lock is a counter, can be locked up to USHRT_MAX
+ * Verified by ::test_lock_eq_ushrt_max and ::test_lock_gt_ushrt_max
+ *
  * This function is IRQ and thread safe
+ * Verified by ::sleep_manager_multithread_test and ::sleep_manager_irq_test
```

Comment:
@0xc0170 Is this way of referring tests acceptable? Or should I rather insert a separate defined/undefined behaviour section, e.g. as seen in the ticker API? https://github.com/ARMmbed/mbed-os/blob/5d5ca62a5e6ee836b86503d58cae6eec8444458f/hal/us_ticker_api.h#L60-L87

Reply:
I assume as its in ticker (defined behavior + how it is verified)

Reply:
I created a new doxygen section for defined behavior.